bindings/go/pkg/whisper/model.go

```diff
@@ -82,12 +89,17 @@ func (model *model) Languages() []string {
 }
 
 func (model *model) NewContext() (Context, error) {
+	// By default, specify the greedy strategy
+	return model.NewContextWithStrategy(SAMPLING_GREEDY)
+}
+
+func (model *model) NewContextWithStrategy(strategy SamplingStrategy) (Context, error) {
```

Comment:
Perhaps instead of passing in just the strategy to this function, it could accepts a parameter of type `whisper.Params`?